### PR TITLE
chore: deprecate fullbleed prop

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [] 

### Description

This PR is a follow-up to previous navigation infra improvements.

In this PR, I work on removing some custom prop that we use for defining screens safe edges: the famous `fullbleed`. This prop although is very intuitive to use. You first need to learn, then you would have to guess that you need to define it with your route **instead** of within the screen itself. Of course that led to it being very coupled to screens layout.

Soo, I removed its usage from the new navigation infra then I went through every screen that doesn't define `fullbleed` around 100 😄 and updated all the ones that broke.

The ideal `viewOptions` props that a screen would need, should be no props other than the default `screenOptions` from `react-navigation`

**Bonus:**
I aligned the navigation headers of so many old screens to match the new patterns 

**Screenshots**
I would like to post screenshots but I don't know if it makes sense to make screen shots to 150 screens - I did though check everything but if anyone raises a broken padding, we will have 2 weeks to fix it

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- deprecate fullbleed - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (deprecate fullbleed - mounir - MounirDhahri)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2e6840dfcb8de1ea128ded77ea18d84065ad6e4c
</p>

**Review by damassi — DISMISSED** [review]
Yess, this one has been a long time coming 👍

**Review by olerichter00 — APPROVED** [review]
Looks great 🌟

File: src/app/routes.ts [hunk 1]
```diff
@@ -265,7 +265,6 @@ export function getDomainMap(): Record<string, RouteMatcher[] | null> {
     addRoute("/feature/:slug", "Feature"),
     addRoute("/galleries-for-you", "GalleriesForYou"),
     addRoute("/gene/:geneID", "Gene"),
-    addRoute("/home-view", "HomeView"),
```
Comment (gkartalis) [inline src/app/routes.ts:268]: 👀👀 intentional?

**Review by gkartalis — APPROVED** [review]
thanks! Looks great

File: src/app/routes.ts (hunk 1 above)
Comment (MounirDhahri) [reply src/app/routes.ts:268]: Yes, we created this route initially to be able to test the old home view and the new one separately. Then we moved to use the same route with a feature flag and we forgot to remove this one.

**Pull request merged**